diesel/src/pg/types/date_and_time/mod.rs

```diff
@@ -143,18 +143,21 @@ impl ToSql<sql_types::Interval, Pg> for PgInterval {
 }
 
 impl FromSql<sql_types::Interval, Pg> for PgInterval {
-    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
-        let bytes = not_none!(bytes);
+    fn from_sql(bytes: Option<&PgValue>) -> deserialize::Result<Self> {
+        let bytes = not_none_pg!(bytes);
         Ok(PgInterval {
-            microseconds: try!(FromSql::<sql_types::BigInt, Pg>::from_sql(Some(
-                &bytes[..8]
-            ))),
-            days: try!(FromSql::<sql_types::Integer, Pg>::from_sql(Some(
-                &bytes[8..12]
-            ))),
-            months: try!(FromSql::<sql_types::Integer, Pg>::from_sql(Some(
-                &bytes[12..16]
-            ))),
+            microseconds: try!(FromSql::<sql_types::BigInt, Pg>::from_sql(Some(&PgValue {
+                data: bytes[..8].to_vec(),
+                oid: 1, // TODO: FIX
```

Comment:
In these cases we don't care since we know it's not looking at the oid. We should probably encode this somehow though if we don't make arbitrary lookup possible here.

Reply:
I was worried about having an Enum for all the possible OIDs, but I'm not sure how feasible that is.

Reply:
OIDs aren't fixed, which is why we have the dynamic lookup in the first place.